packages/nextjs/src/server/utils.ts

```diff
@@ -155,6 +157,7 @@ export function decorateRequest(
       [constants.Headers.AuthMessage]: message || '',
       [constants.Headers.AuthReason]: reason || '',
       [constants.Headers.ClerkUrl]: req.clerkUrl.toString(),
```

Comment:
(not a concern of this PR)

I wonder if we'd like to refactor how we propagate these values using the mechanism Laura introduces in this PR.
We don't need encryption for these, but I think Laura's way is cleaner: create an object and append it to the request, parse the request and get an object back that contains all the values needed without dealing with details (such as the header keys used).

Let's discuss offline about a potential refactor to simplify this :)